yt_dlp/__init__.py

```diff
@@ -490,6 +486,30 @@ def report_conflict(arg1, arg2):
     # this was the old behaviour if only --all-sub was given.
     if opts.allsubtitles and not opts.writeautomaticsub:
         opts.writesubtitles = True
+    # FFmpegRemoveChapters must run before FFmpegMetadataPP
+    remove_chapters_patterns = []
+    if opts.remove_chapters:
+        for regex in opts.remove_chapters:
+            try:
+                remove_chapters_patterns.append(re.compile(regex))
+            except re.error as err:
+                parser.error(f'invalid --remove-chapters regex {regex!r} - {err}')
+        postprocessors.append({
+            'key': 'FFmpegRemoveChapters',
+            'regexes': remove_chapters_patterns,
+            'force_keyframes': opts.force_keyframes_at_cuts,
+            'force': opts.force_remove_chapters,
+        })
+    # FFmpegMetadataPP should be run after FFmpegVideoConvertorPP and
+    # FFmpegExtractAudioPP as containers before conversion may not support
+    # metadata (3gp, webm, etc.)
+    # And this post-processor should be placed before other metadata  <=================== WHAT TO DO ABOUT THIS
+    # manipulating post-processors (FFmpegEmbedSubtitle) to prevent loss of
+    # extra metadata. By default ffmpeg preserves metadata applicable for both
+    # source and target containers. From this point the container won't change,
+    # so metadata can be added here.
```

Comment:
This comment implies that moving addmetadata after embedsubs can cause issues. This will need to be thoroughly tested and we may need to make modifications to ensure all metadata and subs is preserved